src/snowflake/snowpark/session.py

```diff
@@ -1933,6 +1934,10 @@ def write_pandas(
                     + (schema + "." if schema else "")
                     + (table_name)
                 )
+            signature = inspect.signature(write_pandas)
+            if not ("use_logical_type" in signature.parameters):
```

Comment:
Do we need to remove `use_logical_type` explicitly? I feel like we only need the change on line 1955?

Reply:
Per our discussion on Slack, when this parameter is passed in kwargs, Python connector will pass it to PyArrow and hence cause the error there.